faunadb-scala/src/main/scala/faunadb/streaming/BodyValueFlowProcessor.scala

```diff
@@ -0,0 +1,88 @@
+package faunadb.streaming
+
+import java.nio.ByteBuffer
+import java.util
+import java.util.concurrent.{Flow, SubmissionPublisher}
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import faunadb.errors.PermissionDeniedException
+import faunadb.values.{StringV, VSuccess, Value}
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConverters.asScalaIteratorConverter
+import scala.util.Try
+
+private [faunadb] class BodyValueFlowProcessor(syncLastTxnTime: Long => Unit) extends SubmissionPublisher[Value] with Flow.Processor[util.List[ByteBuffer], Value] {
+  private val log = LoggerFactory.getLogger(getClass)
+  private val json = new ObjectMapper
+  private var subscription: Flow.Subscription = null
+  private var subscriber: Flow.Subscriber[_ >: Value] = null
+
+  // We do not request data from the publisher until we have one subscriber
+  override def subscribe(subscriber: Flow.Subscriber[_ >: Value]): Unit = {
+    if (this.subscriber == null) {
+      this.subscriber = subscriber
+      super.subscribe(subscriber)
+      requestOne()
+    } else
+      throw new IllegalStateException("BodyValueFlowProcessor can have only one subscriber")
+  }
```

Comment:
Your explanation makes a lot of sense but it is actually a simpler restriction here.

The constraint is propagated from the new jdk client.

```java
 /**
         * Returns a {@code BodyHandler<Publisher<List<ByteBuffer>>>} that creates a
         * {@link BodySubscriber BodySubscriber}{@code <Publisher<List<ByteBuffer>>>}
         * obtained from {@link BodySubscribers#ofPublisher()
         * BodySubscribers.ofPublisher()}.
         *
         * <p> When the {@code HttpResponse} object is returned, the response
         * headers will have been completely read, but the body may not have
         * been fully received yet. The {@link #body()} method returns a
         * {@link Publisher Publisher<List<ByteBuffer>>} from which the body
         * response bytes can be obtained as they are received. The publisher
         * can and must be subscribed to only once.
         *
         * @apiNote See {@link BodySubscribers#ofPublisher()} for more
         * information.
         *
         * @return a response body handler
         */
        public static BodyHandler<Publisher<List<ByteBuffer>>> ofPublisher() {
            return (responseInfo) -> BodySubscribers.ofPublisher();
        }
```

It says explicitly **The publisher can and must be subscribed to only once.** so I decided to make it invariant at the caller.

Reply:
TBH I now also remember fighting with a race condition that was fixed by introducing this guarding statement.

Reply:
Oh, that's interesting! Yeah, I tested removing the guard and the downstream subscriber didn't not receive the events because the Processor had already consumed them before the subscriber had the chance to subscribe to it.

Some further notes on the comment would be nice (specially for our future selves), but if there's not time for it I would not consider it a blocker.